manifests/map.pp

```diff
@@ -17,23 +19,35 @@
   $mnt = $name
 
   # variable preparations and validations
+  case $mapname {
+    undef:   { $mapname_real = "mountmap_${name}" }
+    default: { $mapname_real = $mapname }
+  }
+
+  case $mappath {
+    undef:   { $mappath_real = "/etc/auto.${name}" }
+    default: { $mappath_real = $mappath }
+  }
+
   case type3x($mounts) {
     'array':  { $mounts_array = $mounts }
     'string': { $mounts_array = [ $mounts ] }
     default:  { fail('autofs::map::mounts is not an array.') }
   }
 
-  if is_string($file) == false { fail('autofs::map::file is not a string.') }
+  if is_string($file)         == false { fail('autofs::map::file is not a string.') }
+  if is_string($mapname_real) == false { fail('autofs::map::mapname is not a string.') }
+  if is_absolute_path($mappath_real) == false { fail('autofs::map::mappath is not an absolute path.') }
 
   # functionality
   $content = $file ? {
     undef   => template('autofs/mountmap.erb'),
     default => undef,
   }
 
-  file { "mountmap_${mnt}":
+  file { $mapname_real:
```

Comment:
This part OS why the code is failing in our environment.

Both puppet-module-afs and puppet-module-autofs is trying to manage `File[home]`.

Reply:
Are only the name conflicting, or actually the file that is managed ?
In first case a fix should be easy to deploy :)

Reply:
It's only the title of the resource!

Reply:
Have changed the resource title so it shouldn't conflict anymore.
Could you please retest if that fix your real life problems now.

Reply:
Yes! Will test tomorrow. Thanks!